src/ng/http.js

```diff
@@ -1088,7 +1103,11 @@ function $HttpProvider() {
         }
 
         // send request
-        return sendReq(config, reqData).then(transformResponse, transformResponse);
+        try {
+          return sendReq(config, reqData).then(transformResponse, transformResponse);
+        } catch (e) {
+          return $q.reject(e);
```

Comment:
This is inside `serverRequest`, which is passed as an "onFulfilled" callback to `promise`: `promise.then(serverRequest)`.
So, even synchronous errors in `sendReq()` will be converted to rejections.